src/main/scala/no/ndla/searchapi/service/search/SearchConverterService.scala

```diff
@@ -767,8 +767,16 @@ trait SearchConverterService {
                                              taxonomyType: String,
                                              bundle: TaxonomyBundle): Try[List[SearchableTaxonomyContext]] = {
       val (resources, topics) = getTaxonomyResourceAndTopicsForId(id, bundle, taxonomyType)
-      val resourceContexts = resources.map(resource => getResourceTaxonomyContexts(resource, bundle))
-      val topicContexts = topics.map(topic => getTopicTaxonomyContexts(topic, bundle))
+      val resourceContexts = resources
+        .filter(resource =>
+          bundle.getSubject(resource.path).exists(subject => subject.metadata.exists(metadata => metadata.visible)))
+        .filter(resource => resource.metadata.exists(metadata => metadata.visible))
+        .map(resource => getResourceTaxonomyContexts(resource, bundle))
+      val topicContexts = topics
+        .filter(topic =>
+          bundle.getSubject(topic.path).exists(subject => subject.metadata.exists(metadata => metadata.visible)))
+        .filter(topic => topic.metadata.exists(metadata => metadata.visible))
+        .map(topic => getTopicTaxonomyContexts(topic, bundle))
```

Comment:
Kanskje dra dette ut i en egen funksjon siden det er likt for resources og topics?
Feks noe sånt?
```
private def filterByVisibility[T <: TaxonomyElement](elementsToFilter: List[T], bundle: TaxonomyBundle): List[T] = {
  val isVisible = (e: TaxonomyElement) => e.metadata.exists(_.visible)
  val hasVisibleSubject = (e: TaxonomyElement) => bundle.getSubject(e.path).exists(isVisible)

  elementsToFilter
    .filter(isVisible)
    .filter(hasVisibleSubject)
}
```